src/backend/distributed/master/worker_node_manager.c

```diff
@@ -332,6 +332,12 @@ ActivePrimaryNodeList(void)
 	HTAB *workerNodeHash = GetWorkerNodeHash();
 	HASH_SEQ_STATUS status;
 
+	if (ReadFromSecondaries == READ_FROM_SECONDARIES_ALWAYS)
+	{
+		ereport(ERROR, (errmsg("writing to nodes is not currently allowed"),
+						errdetail("citus.read_from_secondaries is set to 'always'")));
```

Comment:
I think you need both?

For the current use-case `RecoveryInProgress()` is enough, but if you ever want to toggle `read_from_secondaries` on the primary coordinator this should continue to disallow writes?